src/client/QXmppRosterManager.cpp

```diff
@@ -267,9 +222,8 @@ QXmppTask<QXmppRosterManager::Result> QXmppRosterManager::addRosterItem(const QS
     item.setSubscriptionType(QXmppRosterIq::Item::NotSet);
 
     QXmppRosterIq iq;
-    iq.setType(QXmppIq::Set);
     iq.addItem(item);
-    return client()->sendGenericIq(std::move(iq));
+    return setRoster(std::move(iq));
```

Comment:
Can't recall exactly, but I think it's because else I would not receive the changes in the TestClient for the unit tests.

Reply:
What's the difference between client()->sendGenericIq and client()->sendIq ?
I did read the link, but it does not really tell me particular, arguing my very limited xmpp knowledge

Reply:
`sendGenericIq()` is used in case the exact IQ response is not of interest. It simply returns QXmpp::Success if the IQ request was successful and QXmppError if not. `sendIq()` is used if you need to do process the response.